src/ast/passes/codegen_llvm.cpp

```diff
@@ -2054,6 +2057,33 @@ void CodegenLLVM::visit(Ternary &ternary)
     b_.CreateBr(done);
     b_.SetInsertPoint(done);
     expr_ = nullptr;
+  } else {
```

Comment:
This new branch could be merged with the branch for `Type::string`, couldn't it?